XpringKit/DefaultXpringClient.swift

```diff
@@ -78,6 +80,15 @@ extension DefaultXpringClient: XpringClientDecorator {
   /// - Throws: An error if there was a problem communicating with the XRP Ledger.
   /// - Returns: The status of the given transaction.
   public func getRawTransactionStatus(for transactionHash: TransactionHash) throws -> RawTransactionStatus {
-    throw XRPLedgerError.unimplemented
+    let transactionHashBytes = try transactionHash.toBytes()
```

Comment:
Does the `try` here imply it can fail? What gets stored in `transactionHashBytes` on failure?

Or does that throw an error, and we stop executing (hence the `throws` in the method signature?)

Reply:
On line 81, I've added a `throws` keyword, which tells callers of the function they need to handle an error which may be thrown. 

In this case `transactionHashBytes` is typed as a byte array (`[UInt8]`) rather than an optional byte array (`[UInt8]?`). 

There are three ways to handle errors in Swift:
- `try`: Handle a thrown exception (either by propagating upwards with your method signature, or with a  `catch` block.
- `try?`:  Ignore any thrown exceptions and return `nil` (Swift's version of null) if an exception is thrown. Note that this would cause the type compiler to infer `transactionHashBytes` as an optional byte array, which would change our logic below.
- `try!`: Crash the program if an exception is thrown. This is poor form to use in anywhere that isn't test code.

Swift enforces exception handling at compile time for safety. So if I removed `try` from this line, the program would fail to compile because I haven't handled a possible exception. Likewise, if I removed `throws` from line 82, I would also get a compile time error. This in turn means that anyone who calls this method will have to use `try` and handle exceptions appropriately.